app/models/person.rb

```diff
@@ -37,4 +37,17 @@ def roles_inclusion
       errors.add(:roles,"'#{invalid_role}' is a invalid role.")
     end
   end
+  
+  def find_current_institutions
+	current_affiliations = self.affiliations.where("year_to <= #{Date.today.year} OR year_to IS NULL").all
+	institutions = []
+	current_affiliations.each do |current_affiliation|
+	  institutions << current_affiliation.department.institution.name
```

Comment:
Zostawić `insitution` a nie `name`. Ale lepiej byłoby oddać w afiliacji metodę `institution`, która robi to co trzeba, tzn. przeskakuje przez `department`.